pkg/platform/runtime/setup/events/events.go

```diff
@@ -0,0 +1,204 @@
+package events
```

Comment:
I'm not sure I understand why this package exists. It seems to contain a ton of boilerplate and very little actual logic. Why do we need this?

Reply:
To address your overall criticism of the package organization, let me explain a little more what my motivations were for the organization as it is:

As you saw the logic is split into the `runbits` and the `runtime/setup/events` packages.  
- My intention is to keep the generation of the events in the `events` package, because I felt that this has nothing to do with "code that prints"
- The same is true for the `RuntimeEventConsumer` that consumes the events, and carries and manipulates  some state with it.  - As a result, the `ProgressBar` in the `runbits` package does not have to deal with events anymore and is more target to producing output, hence it is in `runbits`
- If we want to do our own progress bar implementation, we could just replace the `runbits/progressbar` implementation with something that does not depend on `vbauerster/mpb`

So, in short the main distinction is: `runbits` has structs that deal with outputs, and `setup/events` with events as the are reported by the setup update function.

Reply:
Also if you refer to the `RuntimeProducer` for the boilerplate code: It is indeed not doing much more than creating events and sending them to the encapsulated channel.  However, I found it valuable to have a struct that *does* the encapsulation, such that we do not have to expose the channel anywhere.

Writing these simple functions also serves the purpose of implementing interfaces like the `BuildLogMessageHandler` interface.  So, we can easily replace the producer with a mock in tests...